packages/system/src/Location/Components/LocationSearchInput.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { FC } from 'react';
+import {
+  Autocomplete,
+  defaultOptionMapper,
+  getDefaultOptionRenderer,
+} from '@openmsupply-client/common';
+import { useLocations } from '../hooks';
+import { Location } from '../types';
+
+interface LocationSearchInputProps {
+  value: Location | null;
+  width: number;
+  onChange: (location: Location | null) => void;
+  disabled: boolean;
+}
+
+export const LocationSearchInput: FC<LocationSearchInputProps> = ({
+  value,
+  width,
+  onChange,
+  disabled,
+}) => {
+  const { data, isLoading } = useLocations();
+
+  return (
+    <Autocomplete<Location>
+      disabled={disabled}
+      width={`${width}px`}
+      clearable={false}
+      value={
+        value && {
+          ...value,
+          label: value.name,
+        }
+      }
+      loading={isLoading}
+      onChange={(_, location) => {
+        onChange(location);
+      }}
+      options={defaultOptionMapper(data?.nodes ?? [], 'name')}
+      renderOption={getDefaultOptionRenderer('name')}
```

Comment:
Possibly should change this up to be something like `${code} -- ${name}`

Reply:
this looks ok to me - probably because the test locations we have a pretty simple ( A1 with code of A1! )

Reply:
Are ya saying to do this or to leave it as is?

Reply:
heh, pretty vague. Leave as is.